src/protocol/sort/generate_permutation.rs

```diff
@@ -145,10 +146,19 @@ pub async fn generate_permutation<F>(
     ctx: SemiHonestContext<'_, F>,
     sort_keys: &[Vec<Replicated<F>>],
     num_bits: u32,
+    malicious_setting: bool
 ) -> Result<Vec<Replicated<F>>, Error>
 where
     F: Field,
 {
+    let mut malicious_validator = None;
+    let sort_ctx: &dyn Context<F> = if malicious_setting {
```

Comment:
@akoshelev need your help in checking how to make this work. It would be nice to have malicious and semi honest sort work from the same function since we can avoid duplication and mistakes in upgrades by this approach.